### ISSUE TITLE
Sprint May 1 to May 12

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#15253]
# Global Sprint Planning

[Last sprint's issue](https://github.com/PostHog/posthog/issues/15043)

## 3 things that might take us down

1. Clickhouse has been really unhappy for a while @yakkomajuri - already stopped the bleeding but with @ellie still need to upgrade clickhouse and migrate to new cluster where nodes are properly set up - on current sprint
    - Team data warehouse owns some of this medium-term, not this sprint @frankh @fuziontech 
      - Need better backups - current snapshots might not work well
2. Thursday's PG migration 🤞 

Issues from last sprint all resolved or in progress.

## Retro: What can we do better next sprint?

1. Did support form / no support hero work okay?
  - @mariusandra - basically good, but lost capacity in the team because one person on the team is doing support. But issues are more targeted. Good for quality, but not good for capacity. Quality outweighs capacity so overall good.
  - @raquelmsmith (and others) likes it! relevant issues get to me faster.

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
Team ___

Support heros next 2 weeks: 

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by benjackwhite** [comment]
# Team Monitoring

## Retro

- @pauldambra I feel good! Tired brain from thinking more which is good (part of the goals for the team move)
- @annikaschmid wasn't too close to the team due to the technical focus of the sprint (but thats okay)
- @benjackwhite Working with paul is great. Concerned about our Q2 goals. Other than that really enjoying the work. Not enjoying hiring.

## Hang over items from previous sprint

- Get new recording ingestion flow up and running ✅
- Scope out Notebooks, Errors, Automation... 🤔
- Enable the new rrweb version for new teams ❌
- Rename team to "Monitoring" and product to "Replay" ✅

### High priority

-  Notebooks **M**VP - e.g. persistence, usability @benjackwhite and @pauldambra 
why? We want to validate interest in the idea ASAP with real users

### Low priority / side quests

- Observe Recording ingestion to new S3/CH setup on prod

**Comment by mariusandra** [comment]
# Team Product Analytics

Captain Toads next 2 weeks:
- @yakkomajuri first week
- @Twixes second week

## Retro

- Persons on Events v2 ready for rollout
  - Small delay with PoEv2 on HogQL (nothing blocking, needs more time)
  - Cohorts, actions, etc still to do as well.
- PostHog 3000 insight editor
  - Hang over, might not start yet this week. Support has been taking more time. Data exploration loose ends as well.

## Hang over items from previous sprint

- PostHog 3000
- HogQL


## OKR

- Objective 1: PostHog 3000 Hacker News launch is a wild success
  - Ongoing

- Objective 2: Make PostHog performance frustration free for our 10 largest customers
  - Ongoing (PoEv2)

- Objective 3: HogQL
  - Ongoing (Pen test, will do a public beta right after)


### High priority

- "Data Exploration" aka "JSON insights" aka "Name Pending" public launch @thmsobrmlr 
  - *Why?* We're building a unified data structure that describes any data or its visualisation in the frontend.
  - Getting it working for our team
  - Getting it working for all teams
  - Removing old code from the frontend (prerequisite for **PostHog 3000**)
  - Moving the "old -> new -> old migration" layer into the backend
- Get PoEv2 out - @yakkomajuri 
  - *Why?* So that insights for our largest customers calculate in a reasonable amount of time.
- HogQL public beta @mariusandra 
  - *Why?* So that people could use SQL to query their data. 50+ users in the private beta, time to extend it.
  - Waiting for pen test results
  - Misc odds and ends to clean up.

### Low priority / side quests

- PostHog 3000 - @Twixes 
  - Getting Data Exploration out first as a blocker. Then full gear on this.

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro



- Responding to issues with the new billing-by-products API launch @raquelmsmith @charlottemcginn 
  - Went well, no more issues at this point
  - Still need to deploy the frontend
- Free trial experience for CS by @charlottemcginn 
  - Made it work via Django admin again
- Reading available features from the billing service based on products
  - Nearly done, finishing up some work to observe usage limits
  - Should be done before this sprint starts

## Hang over items from previous sprint

- Move usage reports to Temporal @charlottemcginn - decided to deprioritize this as reports have been stable
- Make platform features more understandable/visible in billing-by-products UI @raquelmsmith - will continue for this sprint

## OKR

1. We can easily run experiments on pricing 🟢
2. Each paid product is positioned as a first-class citizen within our onboarding and upgrade flows ⚪ (have not started, so not sure)

### High priority

- [x] Make platform features more understandable/visible in billing-by-products UI @raquelmsmith
- [x] Run experiment with new frontend for subscribe-by-products to make sure it doesn't kill conversion @raquelmsmith
- [ ] Improvements for CS @charlottemcginn 
  - Free trials - is usage still tracked when they are on a free trial? What happens when they go over the usage limits for the plan they are on?
  - Annual flat fees with overage fees - cumbersome to administer, need multiple subscriptions
- [x] Billing limits are applied before discounts but should be applied after @charlottemcginn

### Low priority / side quests

- [ ] Demo data generation OOM fix @charlottemcginn
- [ ] If we have time and experiment results are good, move all customers to new subscribe-by-products UI

**Comment by danielxnj** [comment]
# Team Infrastructure

Infra hero next two weeks @ellie (Ellie will be out from the 3rd of May until the 11th of May for PTO and public holidays, @danielxnj will cover the Infra Hero)

## Retro



- Could not switch Cloudfront as some customer are still sending the wrong Host header and we do not want to drop their data.

## Hang over items from the previous sprint



- US Production Postgres switchover will be done Thursday, April 27th.
- US Switch DNS to Cloudfront will be done Thursday, April 27th.

## OKR

1. AWS Cost 🔴 increase in about 50%,
2. AWS US prod migration from root account to dedicated account 🟡 , Postgres and Kafka are happening next sprint.

### High priority

- Work on AWS costs, as they are going up by about 50%
- Kafka US Production AWS account switch
- Get ingestion working on S3 backed cluster, performance testing, and then add a feature flag to switch over our team to read from the new cluster.

### Low priority/side quests

- Add Clickhouse data source to the internal Grafana

**Comment by neilkakkar** [comment]
## Team Feature Success

Support hero: Eric

### Retro

1. Release Beta management to 5 happy users - [Eric Duong](mailto:eric@posthog.com)
Why? Closing loops from hackathon. Gauge user interest in features.

Not released to users, but should be released internally by Friday.

### Planning 

1. Beta management: Managing rollout & gathering feedback & doing user interviews [Eric Duong](mailto:eric@posthog.com)
	Why? Follow up on what we’ve built
2. Dashboard templates on flags [Eric Duong](mailto:eric@posthog.com)
Why? Core component of feature success. Quick & easy access to flag based analytics.
3. Flag autocapture [Neil Kakkar](mailto:neil@posthog.com)
Why? Core component of feature success. Automates flag analytics & linking to recordings.
4. MVP for productising interviews [Li Yi Yu](mailto:li@posthog.com)
Why? Core component of feature success. Allows getting qualitative user feedback quickly.

**Comment by tiina303** [comment]
# Team Pipeline

Support heros next 2 weeks:  Harry, then Tomas

## Things that might take us down
CH unhappiness
kafka - running out of partitions, now hot with CPU

## Retro
harry: pretty pleased with work with Xavier, fun and annoying ... more and more cowboy, CH stuff has been tiring.
xavier: adjusting to the ship it culture, nice to pair with Harry, really starting to be down on the plugin server status ... doesn't look like quick wins are doable.
jams: good sprint, a lot more going on now, slowly making progress on all fronts
tomas: enjoyed working on batch exports, not happy about losing Luke as he was taking care of the frontend stuff ... its been hard picking that up.
tiina: not much progress forward on PoE, but glad we discovered a query problem and a bunch of support took time.

Last Sprint planning
- 🔴Ingestion PoE columns always properly set [Tiina Turban](mailto:tiina@posthog.com) 
  - we currently see empty personIDs and I suspect person or group properties might be not set properly too
  - + co-working with Yakko to get the query side out fully for anyone on PoEv1
  - squash job running in prod
- ClickHouse Infa side watering [James Greenhill](mailto:james.g@posthog.com)
  - 🟢US scale-up + read-only users + test recovery
    - Read only user complete (moving web over today)
    - Scale up - partially complete
    - Test recovery - not complete (but will be done)
  - 🟢 [Luke Harries](mailto:luke@posthog.com) - will hook HogQL up with the read-only user from here
     - Will be complete today
- session recordings scalability [Xavier Vello](mailto:xavier@posthog.com) & [Harry Waye](mailto:harry@posthog.com)
  - 🟢 Switch the kafka producer & consumer to librdkafka
     - We’re doing this work on the recordings consumer first, but this work should be transferable to analytics too
  - 🟢 repartition events_plugins_ingestion
  - 🟢 BONUS: cooperative rebalancing is enabled for recordings (should help reduce rollout-induced lag)
  - 🟡 autoscaling on kafka lag with KEDA -> rolled out, but needs tuning
- batch exports to S3 [Luke Harries](mailto:luke@posthog.com) [Tomás Farías Santana](mailto:tomas@posthog.com)
  - [Luke Harries](mailto:luke@posthog.com) for frontend (stretch doing the event streaming UI too in prep for when the backend is ready)
  - [Tomás Farías Santana](mailto:tomas@posthog.com) for backend

## Hang over items from previous sprint



Ingestion PoE columns always properly set - delayed to wait out Postgres migration to Aurora as required changes will impact PG.
Squash not ran yet due to CH unhappiness
Recordings autoscaling tuning (KEDA)


## OKR

- Objective: CDP - Owner @fuziontech
  - 🟡Key Result: Rock solid webhooks - Owner @hazzadous 
  - 🟢Key Result: Rock solid batch processing (exports) - Owner @tomasfarias
- Objective: Pipeline unblocks query performance and ensures data quality - Owner @tiina303
  - 🟡Key Result: Persons on Events write path and squash shipped with monitoring and alerting
      - yellow because we've seen CH be quite unhappy recently and running squash isn't going to make it jump from joy
  - 🟡Key Result: Backfills are complete for all teams
  - 🟡Key Result: Guaranteed deduplication of UUIDs in events table within 7 days (with docs explaining it)
      - jams tested deduping manually on a single month and it worked, but again CH unhappiness could be a blocker here
  - 🟡Key Result: Query time deduplication of events to cover until ClickHouse deduplication is complete
 - Objective: Pipeline Robustness and Reliability - Owner @xvello
   - 🟡 Key Result: Errors cause kafka lag instead of DLQ production e.g. person creation failures cause kafka lag not go to DLQ
       - Work not started for analytics, confident we can ship it this quarter
       - Recordings has better error management thanks to the librdkafka work
   - 🟢 Key Result: Plugin server scales elastically and deploys do not create more than 2 minutes of lag
      - migration to librdkafka is bringing huge improvements (recordings workload only for now)
      - Autoscaling on Kafka lag is rolling out for recordings on prod-us

### High priority

- making sure we don't drop person properties when there are updates across different distinct IDs @tiina303 
   - use the read replica for some of the stuff in plugin server too
   - investigation of empty person_id
- Librdkafka migration for analytics @xvello
   - Contribute nodes worker support to librdkafka
   - Analytics workload uses librdkafa for consumer & producer
- batch exports to S3 enabled for 3 teams

### Low priority / side quests

Brownbag on celled architecture @xvello
Kafka-based autoscaling: tune the hpa for recordings + write runbook @xvello

**Comment by fuziontech** [comment]
## 🪣  Team Data Warehouse 🪣 

Support hero: James

### Retro

- We are new! Exciting to have a team owning CH. 🎉 

### Planning 

#### High Priority

- S3 backed mergetree consuming inserts from Prod successfully (and not falling behind on merges) on K8s
- Investigate querying issues with hitting replicas and not shards with distributed table
- Begin migration to intermediate US production cluster using Terraform/Ansible for configs
- Develop plan for backups with tests on US cluster

#### Side Quests

- Ongoing support for existing issues in Production environments

📈 📈 📈
